File: 107598019/HW2/Software Architecture_MySQL/adapter/controller/DeleteCourseController.cs
using Software_Architecture.adapter.presenter;
using Software_Architecture.usecase;
using Software_Architecture.usecase.deleteCourse;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Software_Architecture.adapter.controller
{
    public class DeleteCourseController
    {
        static CourseRepository courseRepository = new CourseRepositoryImpl();
        public DeleteCourse deleteCourse = new DeleteCourseImpl(courseRepository);
        public DeleteCourseOutput deleteCoursePresenter = new DeleteCoursePresenter();

        public DeleteCourseController() { }
        public DeleteCourseController(string title)
        {
            deleteCourseExecute(title);
        }

        public void deleteCourseExecute(string title) {
            DeleteCourseInput input = (DeleteCourseInput) deleteCourse;
            input.setTitle(title);

            
            DeleteCourseOutput output = deleteCoursePresenter;
            
            deleteCourse.execute(input, output);
        }
    }
}

File: 107598034/HW2/CourseSystem/src/Adapter/Controller/DeleteCourse/DeleteCourseController.java
package Adapter.Controller.DeleteCourse;

import java.util.ArrayList;
import java.util.List;

import Adapter.Controller.ClassDataMapper;
import Adapter.Controller.CourseData;
import Adapter.Presenter.DeleteCoursePresenter;
import Adapter.Repository.InJsonCourseRepository;
import CourseUseCase.CourseRepository;
import CourseUseCase.DeleteCourseUseCase.DeleteCourseInput;
import CourseUseCase.DeleteCourseUseCase.DeleteCourseOutput;
import CourseUseCase.DeleteCourseUseCase.DeleteCourseUseCase;
import CourseUseCase.DeleteCourseUseCase.DeleteCourseUseCaseImpl;

public class DeleteCourseController {
	CourseRepository inJsonRepository;
	
	public void deleteCourse(Object[] courseData) {
		List<CourseData> courseList = new ArrayList<>();
		ClassDataMapper classDataMapper = new ClassDataMapper();
		courseList = classDataMapper.transfromCourseData(courseData);
		
		inJsonRepository = new InJsonCourseRepository();
		DeleteCourseUseCase deleteCourseUseCase = new DeleteCourseUseCaseImpl(inJsonRepository);
		
		DeleteCourseInput input = (DeleteCourseInput)deleteCourseUseCase;
		input.setCourseId(courseList.get(0).getCourseId());
	
		DeleteCourseOutput output =  new DeleteCoursePresenter();
		
		deleteCourseUseCase.execute(input, output);
		
	}
}
